### PR TITLE
Rename payload ExcessBlobs to ExcessDataGas

### PR DESCRIPTION
**Pull request by Inphi**
This is a mechanical refactor and doesn't affect consensus rules.

### Testing
-  `bazel test //beacon-chain/blockchain/...`
- ` bazel test //beacon-chain/rpc/...`
- ` bazel test //consensus-types/...`

### CODE REVIEW

**Review by roberto-bayardo — APPROVED** [review]

**Review by dgcoffman — APPROVED** [review]

File: proto/engine/v1/execution_engine.proto [hunk 1]
```diff
@@ -75,7 +75,7 @@ message ExecutionPayload4844 {
 	bytes base_fee_per_gas      = 12 [(ethereum.eth.ext.ssz_size) = "32"];
 	bytes block_hash            = 13 [(ethereum.eth.ext.ssz_size) = "32"];
 	repeated bytes transactions = 14 [(ethereum.eth.ext.ssz_size) = "?,?", (ethereum.eth.ext.ssz_max)  = "1048576,1073741824"];
-	uint64 excess_blobs         = 15; // New in EIP-4844
+	uint64 excess_data_gas      = 15; // New in EIP-4844
```
Comment (roberto-bayardo) [inline proto/engine/v1/execution_engine.proto]: I guess this should probably be 32 byte byte-field to be compliant with the spec?

Comment (Inphi) [reply proto/engine/v1/execution_engine.proto]: This is fixed.

**Review by roberto-bayardo — APPROVED** [review]
nice to see tests too!

File: proto/migration/v1alpha1_to_v2.go [hunk 2]
```diff
@@ -756,11 +756,11 @@ func BeaconState4844ToProto(st state.BeaconState) (*ethpbv2.BeaconState4844, err
...
 	}
-	var excessBlobs uint64
-	if val, err := sourceLatestExecutionPayloadHeader.ExcessBlobs(); err == nil {
-		excessBlobs = val
+	var excessDataGas []byte
+	if val, err := sourceLatestExecutionPayloadHeader.ExcessDataGas(); err == nil {
+		excessDataGas = val
 	}
-	// else ErrUnsupportedGetter, so we ignore its value
+	// else ErrUnsupportedGetter because it's a pre-4844 Payload, so we ignore its value
```
Comment (Inphi) [inline proto/migration/v1alpha1_to_v2.go:763]: hmm, on second thought. This should be an error. A post-4844 beacon state has no business storing payload headers that aren't 4844 compatible (i.e. with a nil `ExcessDataGas`. Will fix.

**Pull request merged**